src/components/skills-quiz/CourseCard.jsx

```diff
@@ -17,18 +17,108 @@ const CourseCard = ({
   const { enterpriseConfig } = useContext(AppContext);
   const { slug, uuid } = enterpriseConfig;
   const partnerDetails = useMemo(() => {
-    const partners = {};
     if (!Object.keys(course).length || !isDefinedAndNotNull(course.partners)) {
-      partners[course.key] = {};
+      return {};
     }
-    partners[course.key] = {
-      primaryPartner:
-        course.partners.length > 0 ? course.partners[0] : undefined,
+    return {
+      primaryPartner: course.partners.length > 0 ? course.partners[0] : undefined,
       showPartnerLogo: course.partners.length === 1,
     };
-    return partners;
   }, [JSON.stringify(course)]);
 
+  const loadingCard = () => (
+    <Card>
+      <Card.ImageCap
+        as={Skeleton}
+        duration={0}
+      />
+
+      <Card.Header
+        title={
+          <Skeleton count={2} data-testid="course-title-loading" />
+        }
+      />
+
+      <Card.Section>
+        <Skeleton duration={0} data-testid="partner-name-loading" />
+      </Card.Section>
+
+      <Card.Section>
+        <Skeleton count={1} data-testid="skills-loading" />
+      </Card.Section>
+    </Card>
+  );
+
+  const courseCard = () => {
+    const primaryPartnerLogo = partnerDetails.primaryPartner && partnerDetails.showPartnerLogo ? {
+      src: partnerDetails.primaryPartner.logoImageUrl,
+      alt: partnerDetails.primaryPartner.name,
+    } : undefined;
+
+    return (
+      <Card isClickable>
+        <Card.ImageCap
+          src={course.cardImageUrl}
+          srcAlt={course.title}
+          logoSrc={primaryPartnerLogo?.src}
+          logoAlt={primaryPartnerLogo?.alt}
+        />
+
+        <Card.Header
+          title={(
+            <Truncate
+              lines={course.skillNames?.length < 5 ? 3 : 2}
+              trimWhitespace
+            >
+              {course.title}
+            </Truncate>
+          )}
+        />
+
+        <Card.Section className="py-3">
+          <>
+            {course.partners.length > 0 && (
+              <p className="partner text-muted m-0">
+                <Truncate lines={2} trimWhitespace>
+                  {course.partners
+                    .map((partner) => partner.name)
+                    .join(', ')}
+                </Truncate>
+              </p>
+            )}
```

Comment:
[question] Is the course partners technically the `Card.Header` `subtitle`?

Reply:
Yup, it looks a little better like that, too.
<img width="325" alt="image" src="https://user-images.githubusercontent.com/2307986/170715363-073d5de8-811c-4eee-b6d6-37b2357230b2.png">